### PR TITLE
Ensure removed hosts ZK node exists as an empty list

### PR DESCRIPTION
**Pull request by negz**
Reviewer: @mattnworb

### CODE REVIEW

File: helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java [hunk 1]
```diff
@@ -531,6 +531,8 @@ public void updateDeploymentGroupHosts(final String groupName, final List<String
       // statusDeploymentGroupRemovedHosts may not exist for deployment groups created before it was
       // introduced.
       client.ensurePath(Paths.statusDeploymentGroupRemovedHosts(groupName));
+      client.ensurePathAndSetData(Paths.statusDeploymentGroupRemovedHosts(groupName),
```
Comment (mattnworb) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: I think if the path is created on the line above, then this call will be a no-op and you won't write the `[]`

Comment (negz) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: Oh yeah. That line was supposed to go away. Fixed.

File: helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/ZooKeeperClient.java [hunk 2]
```diff
@@ -45,6 +45,8 @@
 
   void ensurePath(String path, boolean excludingLast) throws KeeperException;
 
+  void ensurePathAndSetData(String path, byte[] data) throws KeeperException;
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/ZooKeeperClient.java]: Add a javadoc for this to differentiate it from the method above?

Comment (negz) [inline helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/ZooKeeperClient.java]: Done.

**Pull request merged**